Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAsset.cpp

```diff
@@ -245,14 +267,14 @@ namespace AZ
 
             for (auto& shaderItem : m_generalShaderCollection)
             {
-                TryReplaceAsset(shaderItem.m_shaderAsset, asset);
+                shaderItem.TryReplaceShaderAsset(asset);
             }
 
             for (auto& materialPipelinePair : m_materialPipelinePayloads)
             {
                 for (auto& shaderItem : materialPipelinePair.second.m_shaderCollection)
                 {
-                    TryReplaceAsset(shaderItem.m_shaderAsset, asset);
+                    shaderItem.TryReplaceShaderAsset(asset);
                 }
             }
```

Comment:
Can these loops be moved into a function on the shader collection class?

ShaderCollection::TryReplaceShaderAsset

Reply:
Will do.

Reply:
Done.